compiler/modules/types/list.bal

```diff
@@ -34,12 +34,25 @@ public function listAtomicTypeMemberAt(ListAtomicType atomic, int i) returns Sem
 
 public type ListMemberTypes [Range[], SemType[]];
 
-public function listAtomicTypeAllMemberTypes(ListAtomicType atomicType) returns ListMemberTypes {
+function listMemberTypesUnion(ListMemberTypes mt1, ListMemberTypes mt2) returns ListMemberTypes {
+    var [r1, t1] = mt1;
+    var [r2, t2] = mt2;
     Range[] ranges = [];
     SemType[] types = [];
-    SemType[] initial = atomicType.members.initial;
+    foreach var [r, i1, i2] in combineRanges(r1, r2) {
+        ranges.push(r);
+        types.push(union(i1 == () ? NEVER : t1[i1],
+                         i2 == () ? NEVER : t2[i2]));
+    }
+    return [ranges, types];
+}
+
+public function listAtomicTypeAllMemberTypes(FixedLengthArray fixedArray, SemType rest) returns ListMemberTypes {
```

Comment:
This signature change is not OK: the function name starts with `listAtomicType` because it operates on a ListAtomicType. No public functions take separated FixedLengthArray/SemType arguments.